### ISSUE TITLE
Mass Departure Excel Upload (Jen)

### ISSUE THREAD

**Issue by chris206** [206Tours/DB#1999]
New person, Jen has an amazing idea:

We upload an obscene amount of data anytime we put new dates into the system.
This is monotonous, already exists in digital format, and is totally unnecessary.

Instead of creating new departure after new departure by hand and saving each (which takes forever and is less likely to be accurate, ironically), we need a way to upload all of the data in a single excel sheet and have many departures created instantaneously.

You give us the template, we will add the data. You can use just the columns below in your template (to keep it pertinent), or all potential lines of data and we will just write around ones that arent needed.

Keep all defaults as is unless we overwrite

![image](https://cloud.githubusercontent.com/assets/22382144/24268013/9ca726f6-0fe2-11e7-8e2f-699a082e853d.png)

**Comment by bradatkins** [comment]
This already exists and Sandy has used it before:

![image](https://cloud.githubusercontent.com/assets/8440255/24269032/1f5b5970-0fe6-11e7-91c1-71a845d96a94.png)

**Comment by chris206** [comment]
what the freaking heck.

What does the template look like? Can you attach it for download within the import feature?

**Comment by bradatkins** [comment]
http://www.206tours.com/register/uploads/1upload2014.xls

Not sure it's been used in a while, so it may have to be tweaked to add new fields, but don't think that'd be a big deal.

**Comment by chris206** [comment]
ok, great - lets update this

We would leave the Tour ID blank and it would auto-associate, right?

**Comment by chris206** [comment]
Also - can you include the link on the upload page? This will make sure that it never gets lost again.
Not this link, the new one

**Comment by bradatkins** [comment]
Yes, Tour Departure ID is typically left blank unless you want to mass update.

**Comment by bradatkins** [comment]
I won't add a link, because the best thing to do is do an export of existing tours as your starting point, modify, and then upload.

Suggest testing with one row first :)

**Comment by chris206** [comment]
Ok, i see.

Could I just export current format rather than tours? So i just download a blank document with all of the formatting but nothing built in to confuse and screw me up.

**Comment by bradatkins** [comment]
No you can't, but it's good to have examples so  you can see what the
format is. You can just clip the first row (headers) into a different excel
file.

It's important to keep the header row and column order.

**Comment by chris206** [comment]
a couple of issues with that:

1. Only admin has access to all departure export
2. It is not advisable to allow regular users to export all pricing and group leader info - security concern
3. All they need is the header to export in a blank excel document - that has to be the easiest thing to program!

On the import screen, would it be so hard to have a link that opens an excel document with the necessary header? People have been to the moon!!!! hahah

It would eliminate the need to ask every time there is a new column.

**Comment by chris206** [comment]
OK - so Jen tried to upload that file and it just hung there.

Apparently, they are about to start ading tours in the system for the next 2-3 years, so a basic functioning feature would be great

![image](https://cloud.githubusercontent.com/assets/22382144/24665651/0a348b94-192c-11e7-8b7a-92d3ac66fd4d.png)

**Comment by bradatkins** [comment]
Can you email me the file they are using?

**Comment by chris206** [comment]
There seems to be an ability to export a "New Sheet". Is that new?
But it exports a corrupt file with information pre-loaded. Any ideas?

![image](https://cloud.githubusercontent.com/assets/22382144/24665750/5169ffc6-192c-11e7-8d16-945be1d097f2.png)

![image](https://cloud.githubusercontent.com/assets/22382144/24665761/57211ed6-192c-11e7-9aaf-8ddc01b7a3f8.png)

**Comment by bradatkins** [comment]
It is not new, nor is it actually 'corrupted'. Just click Yes and proceed

**Comment by chris206** [comment]
I did and it had info in it already. Should I just tell her to blank that out and fill it in
Then try again?

**Comment by chris206** [comment]
It also tries to save as a "web file" instead of Excel Worksheet

**Comment by chris206** [comment]
ok, you're the man. You just saved jen countless hours

**Comment by bradatkins** [comment]
So it's working?

**Comment by chris206** [comment]
apparently so. She needs to go over each with a fine tooth comb

**Comment by chris206** [comment]
ok, it doesnt work now - watch what Jen does at 5 oclock today

**Comment by bradatkins** [comment]
I need the import file, please email it
